google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -186,12 +190,12 @@ public static Timestamp parseTimestamp(String timestamp) {
                 && timestamp.length() > 20
                 && (timestamp.charAt(20) != 'Z' || timestamp.charAt(20) == 'z')),
         invalidTimestamp);
-    int year = IntParser.parseInt(timestamp, 0, 4);
-    int month = IntParser.parseInt(timestamp, 5, 7);
-    int day = IntParser.parseInt(timestamp, 8, 10);
-    int hour = IntParser.parseInt(timestamp, 11, 13);
-    int minute = IntParser.parseInt(timestamp, 14, 16);
-    int second = IntParser.parseInt(timestamp, 17, 19);
+    int year = Integer.parseInt(timestamp.substring(0, 4));
```

Comment:
I think you need to catch `NumberFormatException`.

Reply:
I don't think that would add very much. The method already throws `IllegalArgumentException` if the input is not valid. `NumberFormatException` is a subclass of `IllegalArgumentException`, so it should be caught by any exception handling the client already has.
The only reason to do so would be to add a custom message. Is that what you mean?